File: game-logic-core/scala3MainProject/src/main/scala/fwc/game/planningPhase/Order.scala
package fwc.game.planningPhase

import fwc.JsonSerializable
import scala.util.Try

case class Order(orderType: OrderType, isStar: Boolean = false, modifier: Int = 0) extends JsonSerializable {
  
  def toJson: ujson.Value = {
    val order = ujson.Obj(
      "type" -> ujson.Str(orderType.toString)
    )
    if isStar then order.obj addOne ("isStar" -> ujson.Bool(isStar))
    if modifier != 0 then order.obj addOne ("modifier" -> ujson.Num(modifier))
    order
  }

}

object Order {
  def fromJson(json: ujson.Value): Order = {
    Order(
      OrderType.fromString(json("type").str),
      Try[Boolean](json.obj("isStar").bool).getOrElse(false),
      Try[Double](json.obj("modifier").num).getOrElse(.0).toInt
    )
  }
}

File: bots/server/game_state/order.py
from server.game_state.order_type import OrderType


class Order:
    def __init__(self, order_type: OrderType, is_star: bool = False, modifier: int = 0):
        self.order_type = order_type
        self.is_star = is_star
        self.modifier = modifier

    @classmethod
    def from_json(cls, json):
        return cls(
            OrderType.from_str(json['type']),
            json['isStar'] if 'isStar' in json else False,
            json['modifier'] if 'modifier' in json else 0
        )